### PR TITLE
QA for dynamic fee estimation - closes #105

### LINKED ISSUE #105: Quality assurance of dynamic fee algorithm
### Description

This issue concerns the quality assurance of the dynamic fee algorithm implemented in the issue https://github.com/LiskHQ/lisk-service/issues/35.

### Motivation

- This feature needs to be well tested before going to production
- The algorithm might cause a lot of difficulties in the implementation phase and need to be properly tested

### Acceptance Criteria

- Bugs are fixed
- Unit tests are written
- Functional tests are written
- Tests are running on Jenkins

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #105 

### How was it solved?

- [x] Bugs are fixed
- [x] Unit tests are written
- [x] Functional tests are written
- [x] Unit test execution on Jenkins CI pipeline is enabled

### How was it tested?
Locally + Jenkins

### CODE REVIEW

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		faker: 'random.number',
+	},
+	maxHeightPreviouslyForged: {
+		faker: 'random.number',
+	},
+	maxHeightPrevoted: {
+		faker: 'random.number',
+	},
+	numberOfTransactions: {
+		function: () => Math.floor(Math.random() * 1000) % 135,
```
Comment (MichalTuleja) [inline services/core/tests/blockGenerator/createBlockHeaderSchema.js:48]: Storing it as a const wouldn't hurt.

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	},
+	payloadHash: {
+		function: () => generateHex(64),
+	},
+	payloadLength: {
+		function: () => Math.floor(Math.random() * 1000),
+	},
+	previousBlockId: {
+		function: () => Math.floor(Math.random() * 10 ** 19),
```
Comment (MichalTuleja) [inline services/core/tests/blockGenerator/createBlockHeaderSchema.js:57]: Please make a comment that this is a top length of that property.

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	},
+	reward: {
+		function: () => String(1 * 10 ** 8),
+	},
+	seedReveal: {
+		function: () => generateHex(32),
+	},
+	timestamp: {
+		function: () => Math.floor(Date.now()) % 4294967295,
```
Comment (MichalTuleja) [inline services/core/tests/blockGenerator/createBlockHeaderSchema.js:66]: Please write a comment about it.

File: services/core/tests/blockGenerator/generateMockData.js [hunk 4]
```diff
@@ -44,10 +44,10 @@ const blockData = {
 		function: () => 1,
 	},
 	timestamp: {
-		function: () => Math.floor(Date.now() / 10) * 10,
+		function: () => Math.floor(Date.now() / 1000),
 	},
 	generatorAddress: {
-		function: () => `${Math.floor(Math.random() * 10 ** 20)}L`,
+		function: () => `${Math.floor(Math.random() * 10 ** 19)}L`,
```
Comment (MichalTuleja) [inline services/core/tests/blockGenerator/generateMockData.js:50]: Please write a comment that we don't cover the whole range of the address system here.

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: services/core/tests/blockGenerator/generateMockData.js (hunk 4 above)
Comment (sameersubudhi) [reply services/core/tests/blockGenerator/generateMockData.js:50]: We are covering the entire range of the address system. A valid Lisk address is a maximum of 20 characters long with the suffix `L` included.

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js (hunk 1 above)
Comment (sameersubudhi) [reply services/core/tests/blockGenerator/createBlockHeaderSchema.js:48]: Done!

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js (hunk 2 above)
Comment (sameersubudhi) [reply services/core/tests/blockGenerator/createBlockHeaderSchema.js:57]: Done!

File: services/core/tests/blockGenerator/createBlockHeaderSchema.js (hunk 3 above)
Comment (sameersubudhi) [reply services/core/tests/blockGenerator/createBlockHeaderSchema.js:66]: Done!

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**